keyboards/primekb/prime_e/keymaps/madhatter/keymap.c

```diff
@@ -96,17 +85,17 @@ void matrix_scan_user(void) {
 
 //function for layer indicator LED
 layer_state_t layer_state_set_user(layer_state_t state) {
-    if (layer_state_cmp(state, 1)) {
+    if (layer_state_cmp(state, 0)) {
         writePinHigh(CAPS_LED_PIN);
     } else {
         writePinLow(CAPS_LED_PIN);
     }
```

Comment:
This could be simplified to: 
```suggestion
    writePin(CAPS_LED_PIN, layer_state_cmp(state, 0));
```

Also, layer 0 may show up as always on, due to how it's handled.